### ISSUE TITLE
Bug in testOpenMMEngine.test_snapshot_set

### ISSUE THREAD

**Issue by dwhswenson** [openpathsampling/openpathsampling#401]
File this under "maybe it doesn't matter", but we seem to be incompatible with numpy 1.9.2. This change happened recently: when I updated with a recent version of master, one of the nosetests failed. To double check it, I just ran tests in the current `master`. I get:

```
ERROR: openpathsampling.tests.testopenmmengine.testOpenMMEngine.test_snapshot_set
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/dwhs/GoogleDrive/msm-tis/openpathsampling/tests/testopenmmengine.py", line 102, in test_snapshot_set
    velocities=np.array(testvel) * u.nanometers / u.picoseconds
  File "/Users/dwhs/GoogleDrive/msm-tis/openpathsampling/netcdfplus/proxy.py", line 128, in _init
    _super_init(self, *args, **kwargs)
  File "/Users/dwhs/GoogleDrive/msm-tis/openpathsampling/snapshot.py", line 381, in __init__
    topology=topology
  File "/Users/dwhs/GoogleDrive/msm-tis/openpathsampling/snapshot.py", line 89, in __init__
    if np.any(np.isnan(coords)):
TypeError: ufunc 'isnan' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
```

I haven't updated any MacPorts stuff on my laptop since before I left Amsterdam, so it will take a chunk of today to *prove* that the problem is with an old numpy version, but I think it likely. If the problem is outdated version of numpy, we might be able to just ignore it, as long as it works in the newer version. 

Assigning to @jhprinz : I thought you might be able to quickly check whether there's anything being cast weirdly as a part of that test (I think that was one of yours, right?). If you don't see a quick fix, feel free to close as a "won't fix," on the basis that the solution is (probably) to upgrade numpy. 

Mainly, this issue serves as documentation in case a future user encounters the problem.

**Comment by dwhswenson** [comment]
Got numpy 1.10 installed. That doesn't fix it. Still updating everything else.

Was there a change in simtk.units that might have made a difference? I'm still on the old one, since I install OpenMM from the website.

**Comment by jchodera** [comment]
There have been some recent changes to `simtk.unit` 25 days ago:
https://github.com/pandegroup/openmm/tree/master/wrappers/python/simtk/unit

**Comment by dwhswenson** [comment]
The changes from 25 days ago are only in docstrings (I read them over -- didn't just trust the commit message). So that's not it.

I also checked the more significant changes from 5 months ago. That looks like it was clean-up to have better code style and better import behavior. I didn't see anything in there that would have changed functionality, although there were a lot of changes and I might have missed something.

Still upgrading stuff, so maybe that will fix it. I'll also take a look at the test itself: everything seems to work except this one test, so there might just be something weird there. I just ran `alanine.ipynb` with no problems. That uses the OpenMM engine and, if setting the snapshot was really the problem, it would fail in an explosion of errors.

My guess is that the test involves some edge case that the conda versions of stuff treat differently than my outdated MacPorts versions of stuff. Of course, now that I'm updating, I have to deal with the reason I didn't update in the first place -- I knew this was going to break my syntax completion in a way that temporarily renders my editor completely unusable. :-/

**Comment by jhprinz** [comment]
Really weird. Here my `conda list` result
```
conda                     3.18.6                   py27_0  
h5py                      2.5.0               np110py27_4  
hdf5                      1.8.15.1                      2  
libnetcdf                 4.3.3.1                       1  
mdtraj                    1.5.1               np110py27_0  
netcdf4                   1.1.9                np19py27_0  
nose                      1.3.7                    py27_0  
numpy                     1.10.1                   py27_0  
openmm                    6.3                      py27_0  
openpathsampling (/Users/jan-hendrikprinz/Studium/git/openpathsampling) 0.0.1a0                   <pip>
python                    2.7.10                        2  
```
A few ideas
1. `copy.deepcopy` on a simtk.unit(np.array) who knows...
2. Have you changed the pdb? Maybe that returns some weird inf or nan values that are not testable? Also masked arrays could be a problem.
3. The pdb reader from mdtraj has changed? What is your version?

Any luck?

**Comment by dwhswenson** [comment]
@jhprinz : The order of how things broke: (1) I switched to the exact current `choderalab/master`; nosetests failed. (2) I starting upgrading things, thinking that maybe something you had changed required a more recent version of one of the packages than what I'd had since July.

So I can't have a different PDB (unless GitHub doesn't recognize it as changed, which would seem odd). I checked with with `git diff upstream/master`. Since the code is identical to master, so the problem seems like it should be in a package version, or, possibly, some weird file left around my directories. I've removed `openpathsampling/*pyc` and `openpathsampling/*/*pyc`. 

Updated numpy to 1.10 (from 1.9), and with it Python from 2.7.9 to 2.7.10: no fix. After seeing your post above, updated mdtraj to 1.5.1 (from 1.4.2): no fix.

Current versions of things you listed:

```
# (obtained from `package.version.full_version` if available, otherwise `package.__version__`)
numpy 1.10.1
netCDF4 1.2.1
simtk.openmm 6.1.0.dev-c4acc7d
mdtraj 1.5.1
nose 1.3.7

# obtained from `port installed`
  hdf5 @1.8.16_0+cxx (active)
  netcdf @4.3.3.1_4+dap+netcdf4 (active)
  python27 @2.7.10_3+universal (active)
```

My OpenMM version is different from yours (and I'm not sure how I ended up with a dev version; I thought I had given up on trying to install it from source and went with the pre-packaged binary). So maybe I should try reinstalling OpenMM. 

Still, this is a weird problem. Something that worked before broke when I updated master.

I'm pretty sure it has to do with this change: https://github.com/choderalab/openpathsampling/commit/f6a941565977c5d18302e6493f42b5f6d703ea7c#diff-6250b3e3ea6d7715ad102aa6738f19b8R100

These are the lines in the test that now break, and they worked before I pulled in these changes. When I revert them back to taking unitless values, the nosetests pass.

I have no idea what has changed in yours such that numpy can check `isnan` on values with units attached, but it looks to me like that's the difference.

**Comment by jhprinz** [comment]
I use the same code and it works fine on my machine. I also checked each part in ipython separately.

Could you add a 
```
print type(coordinates), coordinates.dtype
```
etc. Just to figure out what is going on?

I am confused since my master passes tests...

**Comment by jhprinz** [comment]
New thought. Is it possible that we somehow changed tests and that the number type is an int and not a float? I somehow recall that `NaN` is only supported for some types by specifiying that a specific value should be interpreted as `NaN`

Is this still an issue?

**Comment by jhprinz** [comment]
Can this be closed or is it still an issue?

**Comment by dwhswenson** [comment]
Still an issue for me; but might just be due to out-of-date OpenMM. I haven't had a chance to upgrade OpenMM yet.

**Comment by jhprinz** [comment]
Did you try the `print type(coordinates), coordinates.dtype` to make sure that it is not OPSs fault?

**Comment by jhprinz** [comment]
Solved by update of openmm > 6.3
